### PR TITLE
Add tests for ValidateBasic methods

### LINKED ISSUE #2740: Write more tests for ValidateBasic methods
#2711 introduced ValidateBasic methods for more data structures and reactor messages.

We should write extensive tests for all of these.

### PR DESCRIPTION
**Pull request by katakonst**
Fixes  #2740
* [x] Wrote tests

### CODE REVIEW

File: evidence/reactor_test.go [hunk 1]
```diff
@@ -178,3 +179,45 @@ func TestReactorSelectiveBroadcast(t *testing.T) {
...
+	}{
+		{"Good EvidenceListMessage", func(evList *EvidenceListMessage) {}, false},
+		{"Invalid EvidenceListMessage", func(evList *EvidenceListMessage) {
+			evList.Evidence = append(evList.Evidence,
+				&types.DuplicateVoteEvidence{PubKey: secp256k1.GenPrivKey().PubKey()})
+		}, true},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.testName, func(t *testing.T) {
+			evList := sendEvidence(t, reactors[0].evpool, valAddr, NUM_EVIDENCE)
```
Comment (melekes) [inline evidence/reactor_test.go]: why do we need to send it? can't we just validate it without running reactors and all that stuff?

File: types/heartbeat_test.go [hunk 2]
```diff
@@ -58,3 +60,45 @@ func TestHeartbeatWriteSignBytes(t *testing.T) {
...
+		{"Negative round", func(hb *Heartbeat) {
+			hb.Round = -1
+		}, true},
+		{"Negative sequence", func(hb *Heartbeat) {
+			hb.Sequence = -1
+		}, true},
+		{"Missing signature", func(hb *Heartbeat) {
+			hb.Signature = nil
+		}, true},
+		{"Signature to big", func(hb *Heartbeat) {
```
Comment (melekes) [inline types/heartbeat_test.go]: `to` -> `is too`

Comment (katakonst) [reply types/heartbeat_test.go]: Fixed!

File: evidence/reactor_test.go (hunk 1 above)
Comment (katakonst) [reply evidence/reactor_test.go]: Don't need to run the reactors. 
Deleted that part.

File: evidence/reactor_test.go [hunk 3]
```diff
@@ -178,3 +179,30 @@ func TestReactorSelectiveBroadcast(t *testing.T) {
...
+				&types.DuplicateVoteEvidence{PubKey: secp256k1.GenPrivKey().PubKey()})
+		}, true},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.testName, func(t *testing.T) {
+			evListMsg := &EvidenceListMessage{}
+			n := 3
+			valAddr := []byte("myval")
+			evListMsg.Evidence = make([]types.Evidence, n)
+			for i := 0; i < 3; i++ {
```
Comment (melekes) [inline evidence/reactor_test.go]: 3 -> n

Comment (katakonst) [reply evidence/reactor_test.go]: Changed!

**Pull request merged**